lib/fluent/plugin_helper/socket.rb

```diff
@@ -150,8 +150,19 @@ def socket_create_tls(
           context.ciphers = ciphers
           context.verify_mode = OpenSSL::SSL::VERIFY_PEER
           context.cert_store = cert_store
-          context.verify_hostname = true if verify_fqdn && fqdn && context.respond_to?(:verify_hostname=)
-          context.cert = OpenSSL::X509::Certificate.new(File.read(cert_path)) if cert_path
+          if verify_fqdn && fqdn && context.respond_to?(:verify_hostname=)
+            context.verify_hostname = true
+          else
+            context.verify_hostname = false
+          end
+
+          cert_data = File.read(cert_path)
+          pattern = Regexp.compile('-+BEGIN CERTIFICATE-+\n(?:[^-]*\n)+-+END CERTIFICATE-+\n', Regexp::MULTILINE)
+          client_certs = []
+          cert_data.scan(pattern){|match| client_certs << OpenSSL::X509::Certificate.new(match)}
+          context.cert = client_certs[0]
+          context.extra_chain_cert = client_certs[1..-1]
```

Comment:
these methods are deprecated https://ruby.github.io/openssl/OpenSSL/SSL/SSLContext.html#method-i-add_certificate

> cert=, key=, and extra_chain_cert= are old accessor methods for setting certificate and internally call this method.

how about #add_certificate?

Reply:
oh that's much better, will do